### PR TITLE
Replace Ganache with Hardhat

### PR DESCRIPTION
**Pull request by alexandratran**
Replace Ganache devnet how-to guide with Hardhat.

Preview: https://docs.metamask.io/ganache-to-hardhat/wallet/how-to/get-started-building/run-devnet/

### CODE REVIEW

File: wallet/how-to/get-started-building/run-devnet.md [hunk 1]
```diff
@@ -5,24 +5,40 @@ sidebar_position: 2
...
-## Connect to Ganache
-
-Follow the [Ganache quickstart](https://trufflesuite.com/docs/ganache/quickstart/) to set
-up a development network.
+## Connect to Hardhat
+
+Follow the [Hardhat Quick Start](https://hardhat.org/hardhat-runner/docs/getting-started#quick-start)
+to set up a Hardhat development network.
+
+In your `hardhat.config.js` file, specify your
```
Comment (BelfordZ) [inline wallet/how-to/get-started-building/run-devnet.md:22]: Should have a note here to not use mainnet, or any real seedphrase. People should create a new seed phrase before continueing

File: wallet/how-to/get-started-building/run-devnet.md [hunk 2]
```diff
@@ -5,24 +5,40 @@ sidebar_position: 2
...
+[MetaMask seed phrase](https://support.metamask.io/hc/en-us/articles/360015290032-How-to-reveal-your-Secret-Recovery-Phrase)
+in the [`accounts`](https://hardhat.org/hardhat-network/docs/reference#accounts) field of the
+`networks.hardhat` configuration:
+
+```js title="hardhat.config.js"
+module.exports = {
+  networks: {
+    hardhat: {
+      accounts: {
+        mnemonic: "<your MetaMask seed phrase>",
```
Comment (BelfordZ) [inline wallet/how-to/get-started-building/run-devnet.md:32]: safer to save the seed in your env (export SEED_PHRASE="abc 123"), and use process.env.SEED_PHRASE 

This prevents people from accidently adding their seedphrases to a commit. (hopefully they follow the previous step of creating a new seed phrase, but if they end up using that seedphrase at some point, this is a very easy way to leak your key). 

That being said, I can understand why we may want to give this more simple instruction, so feel free to make the call there.

File: wallet/how-to/get-started-building/run-devnet.md [hunk 3]
```diff
@@ -5,24 +5,40 @@ sidebar_position: 2
...
-When you create a Ganache workspace, enter your MetaMask seed phrase into
-the **Account & Keys** setting.
-Ganache automatically gives each of your first 10 accounts 100 test ether (you can configure
-these numbers in **Accounts & Keys**), which makes it easy to start development.
+Hardhat automatically gives each of your first 20 accounts 10000 test ether (you can modify
+these numbers in the [`accounts`](https://hardhat.org/hardhat-network/docs/reference#accounts)
+configuration), which makes it easy to start development.
 
 :::caution important
 Your seed phrase controls all your accounts, so we recommend keeping at least one seed phrase for
```
Comment (BelfordZ) [inline wallet/how-to/get-started-building/run-devnet.md:44]: ah I see this note here now. I think this should be moved to the top of the section, as its very important that people read this before extracting their seedphrase.

File: wallet/how-to/get-started-building/run-devnet.md [hunk 4]
```diff
@@ -31,20 +47,16 @@ You can manage multiple seed phrases by using multiple browser profiles, each wi
...
+In the MetaMask extension:
 
-1. Select the network you're currently connected to.
+1. In the upper left corner, select the network you're currently connected to.
 1. Select **Add network**.
 1. Select **Add a network manually**.
-1. Enter the RPC URL of your network.
-1. Enter MetaMask's default [chain ID](../connect/detect-network.md#chain-ids) for Ganache, `1337`.
+1. Enter the RPC URL of your Hardhat network, `http://127.0.0.1:8545/`.
+1. Enter MetaMask's default chain ID for Hardhat, `31337`.
```
Comment (BelfordZ) [inline wallet/how-to/get-started-building/run-devnet.md:59]: Couple things here:

1. can we include the hex formatted chain Id as well? (should be `0x7A69` right?)
2. Can we add a pre-populated link to playground for addEthereumChain? Something like this: `https://docs.metamask.io/wallet/reference/wallet_addethereumchain/?AddEthereumChainParameter[rpcUrls][0]=http://127.0.0.1:8545&AddEthereumChainParameter[chainId]=0x7A69&AddEthereumChainParameter[chainName]=Hardhat&AddEthereumChainParameter[nativeCurrency][name]=testEth&AddEthereumChainParameter[nativeCurrency][symbol]=testEth&AddEthereumChainParameter[nativeCurrency][decimals]=18`
3. Some people will have issues with trying to use 127.0.0.1 - We should include the alternative `http://localhost:8545`

**Review by BelfordZ — APPROVED** [review]
Looks great, nice work.

Just had a couple comments, but overall it gets my approval!

File: wallet/how-to/get-started-building/run-devnet.md (hunk 4 above)
Comment (jiexi) [inline wallet/how-to/get-started-building/run-devnet.md:59]: https://hardhat.org/hardhat-network/docs/metamask-issue
https://github.com/MetaMask/metamask-extension/issues/10290

just want to make sure our instructions for adding Hardhat to MetaMask are correct

File: wallet/how-to/get-started-building/run-devnet.md [hunk 5]
```diff
@@ -5,46 +5,84 @@ sidebar_position: 2
 
 # Run a development network
 
-You can run a personal Ethereum development network using [Ganache](https://www.trufflesuite.com/ganache),
+You can run a personal Ethereum development network using [Hardhat](https://hardhat.org/),
 which allows you to develop a dapp in a secure test environment.
 
 :::note
-When using a local development blockchain such as Ganache or
+When using a local development blockchain such as Hardhat or
```
Comment (bgravenorst) [inline wallet/how-to/get-started-building/run-devnet.md:12]: "HardHat Network", is the HardHat component that runs the dev networks. So not sure whether we want to use the component name or platform name. 
https://hardhat.org/hardhat-network/docs/overview#hardhat-network

File: wallet/how-to/get-started-building/run-devnet.md [hunk 6]
```diff
@@ -5,46 +5,84 @@ sidebar_position: 2
...
-```
-
-In the MetaMask extension, connect to your Ganache network:
-
-1. Select the network you're currently connected to.
-1. Select **Add network**.
-1. Select **Add a network manually**.
-1. Enter the RPC URL of your network.
-1. Enter MetaMask's default [chain ID](../connect/detect-network.md#chain-ids) for Ganache, `1337`.
+## Connect to Hardhat
```
Comment (bgravenorst) [inline wallet/how-to/get-started-building/run-devnet.md:17]:
```suggestion
## Connect to Hardhat Network
```

File: wallet/how-to/get-started-building/run-devnet.md [hunk 7]
```diff
@@ -5,46 +5,84 @@ sidebar_position: 2
...
+
+Follow these steps to connect MetaMask to a Hardhat development network.
+
+1. Follow the [Hardhat Quick Start](https://hardhat.org/hardhat-runner/docs/getting-started#quick-start)
```
Comment (bgravenorst) [inline wallet/how-to/get-started-building/run-devnet.md:21]: The quickstart creates a Hardhat project and then compiles, tests, deploys contract etc. So should this step just be, "Create a Hardhat project"? Then link to: https://hardhat.org/hardhat-runner/docs/guides/project-setup

File: wallet/how-to/get-started-building/run-devnet.md [hunk 8]
```diff
@@ -5,46 +5,84 @@ sidebar_position: 2
...
+          },
+          chainId: 1337
+        },
+      },
+    };
+    ```
+
+    Hardhat automatically gives each of your first 20 accounts 10000 test ether (you can modify
+    these numbers in the [`accounts`](https://hardhat.org/hardhat-network/docs/reference#accounts)
+    configuration), which makes it easy to start development.
```
Comment (bgravenorst) [inline wallet/how-to/get-started-building/run-devnet.md:64]:
```suggestion
        ```js title="hardhat.config.js"
        module.exports = {
          networks: {
            hardhat: {
              accounts: {
                mnemonic: "<your MetaMask seed phrase>",
              },
              chainId: 1337
            },
          },
        };
        ```

        Hardhat automatically gives each of your first 20 accounts 10000 test ether (you can modify
        these numbers in the [`accounts`](https://hardhat.org/hardhat-network/docs/reference#accounts)
        configuration), which makes it easy to start development.
    ```

Comment (bgravenorst) [reply wallet/how-to/get-started-building/run-devnet.md:64]: Added indentation so this aligns with the numbered bullet

File: wallet/how-to/get-started-building/run-devnet.md [hunk 9]
```diff
@@ -5,46 +5,84 @@ sidebar_position: 2
...
+   2. Select **Add network**.
+   
+   3. Select **Add a network manually**.
+   
+   4. Enter the RPC URL of your Hardhat network, `http://127.0.0.1:8545/` (or `http://localhost:8545`).
+   
+   5. Enter the chain ID of your Hardhat network, `1337` (or `0x539` in hexadecimal format).
+
+   :::tip
+   Alternatively, you can add your Hardhat network to MetaMask using
```
Comment (bgravenorst) [inline wallet/how-to/get-started-building/run-devnet.md:82]:
```suggestion
   Alternatively, you can add Hardhat Network to MetaMask using
```

File: wallet/how-to/get-started-building/run-devnet.md [hunk 10]
```diff
@@ -5,46 +5,84 @@ sidebar_position: 2
...
-In the MetaMask extension, connect to your Ganache network:
-
-1. Select the network you're currently connected to.
-1. Select **Add network**.
-1. Select **Add a network manually**.
-1. Enter the RPC URL of your network.
-1. Enter MetaMask's default [chain ID](../connect/detect-network.md#chain-ids) for Ganache, `1337`.
+## Connect to Hardhat
+
+Follow these steps to connect MetaMask to a Hardhat development network.
```
Comment (bgravenorst) [inline wallet/how-to/get-started-building/run-devnet.md:19]:
```suggestion
Follow these steps to connect MetaMask to Hardhat Network.
```

**Review by bgravenorst — COMMENTED** [review]
Left some feedback

File: wallet/how-to/get-started-building/run-devnet.md [hunk 11]
```diff
@@ -5,46 +5,84 @@ sidebar_position: 2
...
+    specifically for development.
+
+    :::caution important
+    Your seed phrase controls all your accounts, so we recommend keeping at least one seed phrase for
+    development, separate from any used to store real value.
+    You can manage multiple seed phrases by using multiple browser profiles, each with its own
+    MetaMask installation.
+    :::
+
+3. In your `hardhat.config.js` file, in the `networks.hardhat` configuration:
```
Comment (bgravenorst) [inline wallet/how-to/get-started-building/run-devnet.md:35]: When testing Hardhat I noticed the `hardhat.config.js` doesn't have any networks defined by default. So should this be, "In your `hardhat.config.js` file, create a `networks` section..."

File: wallet/how-to/get-started-building/run-devnet.md (hunk 8 above)
Comment (alexandratran) [reply wallet/how-to/get-started-building/run-devnet.md]: I think the existing indentation was correct here.

**Review by bgravenorst — APPROVED** [review]
LGTM

**Pull request merged**